test/test_unistd.rs

```diff
@@ -927,6 +927,40 @@ fn test_linkat_file() {
     assert!(newfilepath.exists());
 }
 
+#[test]
+#[cfg(not(any(target_os = "redox", target_os = "haiku")))]
+/// This test is the same as [test_linkat_file], but ensures that two different types can be used
+/// as the path arguments.
+fn test_linkat_pathtypes() {
+    use nix::fcntl::AtFlags;
+
+    let tempdir = tempdir().unwrap();
+    let oldfilename = "foo.txt";
+    let oldfilepath = tempdir.path().join(oldfilename).into_os_string();
+
+    let newfilename = "bar.txt";
+    let newfilepath = tempdir.path().join(newfilename);
+
+    // Create file
+    File::create(oldfilepath).unwrap();
+
+    // Get file descriptor for base directory
+    let dirfd =
+        fcntl::open(tempdir.path(), fcntl::OFlag::empty(), stat::Mode::empty())
+            .unwrap();
+
+    // Attempt hard link file at relative path
+    linkat(
+        &dirfd,
+        oldfilename,
```

Comment:
Is this supposed to be oldfilepath?  If not, then both file paths are of the same type.
```suggestion
        oldfilepath,
```

Reply:
Ah, whoops. Will fix.